### PR TITLE
Add MINIMAL pragma for Additive type class

### LINKED ISSUE #11000: Additive instance without `negate` causes GRPC error 'deadline exceeded'
Minimal example:

https://github.com/lucianojoublanc-da/additivebug

There are two issues
* compiler doesn't warn about incomplete definition of `instance Additive`
* using `-` in the example triggers above-said error, when running `daml test`

### PR DESCRIPTION
**Pull request by akrmn**
This ensures that the compiler gives an error if both `(-)` and `negate` are missing in an instance

Fixes #11000

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by akrmn** [comment]
Perhaps we should add @lucianojoublanc-da's example from https://github.com/lucianojoublanc-da/additivebug as a negative test.

**Comment by akrmn** [comment]
I added an empty `changelog_{begin,end}` block to the commit message to fix one of the build errors, though I'm not sure if this sort of change should be part of the log. I checked the repo history and it seems like similar changes haven't been added to the changelog, though.

**Review by sofiafaro-da — APPROVED** [review]
Looks good, thank you!

I agree, it would be nice ta have a test for this (in `compiler/damlc/tests/daml-test-files`).

**Review by cocreature — APPROVED** [review]
Thanks!

**Comment by cocreature** [comment]
> I added an empty `changelog_{begin,end}` block to the commit message to fix one of the build errors, though I'm not sure if this sort of change should be part of the log. I checked the repo history and it seems like similar changes haven't been added to the changelog, though.

That’s exactly the right thing to do. We enforce a changelog to make sure people consider whether it’s changelog entry worthy and if the answer is that it isn’t then adding an empty changelog is the right course of action and we filter those out.

**Comment by akrmn** [comment]
Just added a test case. As suggested on slack, I looked for other classes in the standard library which might need a `MINIMAL` pragma, but I didn't find any.

File: compiler/damlc/tests/daml-test-files/IncompleteAdditiveInstance.daml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+-- Copyright (c) 2020, Digital Asset (Switzerland) GmbH and/or its affiliates.
+-- All rights reserved.
+
+-- @ERROR range=14:10-14:23; • No explicit implementation for; either ‘-’ or ‘negate’; • In the instance declaration for ‘Additive Expr’
```
Comment (cocreature) [inline compiler/damlc/tests/daml-test-files/IncompleteAdditiveInstance.daml:4]: the unicode might cause issues if it’s rendered differently on Windows (not sure if it is, maybe it’s fine). in that case I’d just start shorten it to `No explicit implementation for; either ‘-’ or ‘negate’`. Together with the range that still tests the right thing.

Comment (akrmn) [reply compiler/damlc/tests/daml-test-files/IncompleteAdditiveInstance.daml:4]: ah good point, I fixed it now. Thanks!

Comment (akrmn) [reply compiler/damlc/tests/daml-test-files/IncompleteAdditiveInstance.daml:4]: ah good point, but it seems to have gone through :)

Comment (cocreature) [reply compiler/damlc/tests/daml-test-files/IncompleteAdditiveInstance.daml:4]: if it goes through even better :)

**Pull request merged**